tests/unit/test_plugin.py

```diff
@@ -295,3 +296,29 @@ def test_local_csp_config_save():
             )
 
             assert get_csp_config(config=local_config) == test_data2
+
+
+def test_local_csp_usage_logging_not_set(caplog):
```

Comment:
Minor readability suggestion... The two levels of with patch(...) here are a little ungainly; might be better to do one as a decorator?

Reply:
I would like to move all `with patch` to decorators, that is in my list of "todos" for this repo,
but wanted to check first if there was any reason no decorators were used
will move that one in this PR and will do a subsequent PR to move the rest of `with patch`

Reply:
Patch vs decorator was probably a matter of personal choice on the part of the original coder...